### PR TITLE
Implement new App Launcher Navigation

### PR DESCRIPTION
**Pull request by jbouder**
## Reference Issues or PRs

N/A

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Screenshots
Screenshot 2024-03-14 at 4 54 02 PM
Screenshot 2024-03-14 at 4 54 13 PM
Screenshot 2024-03-14 at 4 54 24 PM

## Any other comments?
- Added Top Navigation (App Bar) to layout
- Added new Profile Menu to Top Navigation
- Added Side Navigation (Drawer) to layout
- Updated Main area for new navigation
- Updates to underlying jhub_apps service to support consistent layout

### CODE REVIEW

File: ui/src/utils/jupyterhub.ts [hunk 1]
```diff
@@ -1,19 +1,9 @@
...
-    ? window.jhdata
-    : JSON.parse(localStorage.getItem('jhdata') || '{}');
-};
-
-/**
- * Store JupyterHub data in local storage for non-hub pages
- * @param data
- */
-export const storeJhData = (data: JhData) => {
-  localStorage.setItem('jhdata', JSON.stringify(data));
```
Comment (jbouder) [inline ui/src/utils/jupyterhub.ts:16]: Realized we had an endpoint for getting user data, so removed this and replaced with accessing from endpoint and storing in global storage

**Comment by aktech** [comment]
nitpick: If there are too many of them on the sidebar, we can't see the last one:

https://github.com/nebari-dev/jhub-apps/assets/5647941/6a2025fd-8a7b-4dbc-8625-248c0c771181

**Comment by jbouder** [comment]
The plan right now is to put the services under expandable sections, which would be collapsed by default, so that should help. We just don’t have the mechanism yet for categorization. That said, it might be worth mentioning during our design/dev sync this week as something to reconsider.

**Review by aktech — APPROVED** [review]
This looks great! 🎉

**Pull request merged**